src/Symfony/Component/Intl/Languages.php

```diff
@@ -52,11 +52,25 @@ public static function exists(string $language): bool
     /**
      * Gets the language name from its alpha2 code.
      *
+     * A full locale may be passed to obtain a more localized language name, e.g. "American English" for "en_US".
+     *
      * @throws MissingResourceException if the language code does not exist
      */
     public static function getName(string $language, string $displayLocale = null): string
     {
-        return self::readEntry(['Names', $language], $displayLocale);
+        try {
+            return self::readEntry(['Names', $language], $displayLocale);
+        } catch (MissingResourceException $e) {
+            try {
+                return self::readEntry(['LocalizedNames', $language], $displayLocale);
+            } catch (MissingResourceException $e) {
+                if (false !== $i = strpos($language, '_')) {
+                    return self::readEntry(['Names', substr($language, 0,  $i)], $displayLocale);
```

Comment:
May be make it a recursive function and call `self::getName` here? That would resolve the case if you have 2 underscores.

Reply:
i see , conceptually we can have e.g. `en_AU_SomeScript` where we need to try `en_AU` still ... :+1: 

i like to finish #33140 first, and get a confirmation on the target branch :)

Reply:
done.